mlrun/k8s_utils.py

```diff
@@ -232,6 +238,54 @@ def handle_invalid(message):
     return True
 
 
+def sanitize_k8s_objects(
+    k8s_objects: typing.Union[None, K8sObj, K8sObjList, SanitizedK8s],
+) -> typing.Union[dict, list[dict], None]:
+    """Convert K8s objects to dicts. Handles single objects or lists."""
+    api_client = kubernetes.client.ApiClient()
+    if not k8s_objects:
+        return k8s_objects
+
+    def _sanitize_k8s_object(k8s_obj):
+        return (
+            api_client.sanitize_for_serialization(k8s_obj)
+            if hasattr(k8s_obj, "to_dict")
+            else k8s_obj
+        )
+
+    return (
+        [_sanitize_k8s_object(k8s_obj) for k8s_obj in k8s_objects]
+        if isinstance(k8s_objects, list)
+        else _sanitize_k8s_object(k8s_objects)
+    )
```

Comment:
```suggestion
    k8s_objects = mlrun.utils.as_list(k8s_objects)
    return list(map(_sanitize_k8s_object, k8s_objects))
```
more elegant I think.

Reply:
I prefer keeping the current approach since it preserves the input/output type relationship as specified in the type hints. Using as_list() would require additional logic to track the original type, which doesn't add much value here. In your suggested approach, we would always return a list, even when the original input wasn't a list.